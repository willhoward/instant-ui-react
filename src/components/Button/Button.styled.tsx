import styled from "styled-components";
import { ButtonProps } from "./Button.types";
import { useTheme } from "../Theme/Theme";

const { black, lightGray, white } = useTheme();

export const StyledButton = styled.button<ButtonProps>`
  height: 42px;
  border: none;
  padding: 0 12px;
  border-radius: 6px;
  font-size: 14px;
  cursor: pointer;
  background: ${({ primary }) => (primary ? white : black)};
  border: 1px solid ${({ primary }) => (primary ? white : lightGray)};
`;
