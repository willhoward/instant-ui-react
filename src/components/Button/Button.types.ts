import { MouseEventHandler } from "react";

export interface ButtonProps {
  text?: string;
  priority?: "primary" | "secondary" | "tertiary";
  disabled?: boolean;
  onClick?: MouseEventHandler<HTMLButtonElement>;
}
