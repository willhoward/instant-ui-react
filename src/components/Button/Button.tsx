import React from 'react';
import { StyledButton } from './Button.styled';
import { ButtonProps } from './Button.types';

export function Button({
	text,
	disabled,
	priority = 'primary',
	onClick,
}: ButtonProps) {
	return (
    <StyledButton
      type='button'
      disabled={disabled}
      priority={priority}
      onClick={onClick}
		>
      {text}
    </StyledButton>
  );
}
