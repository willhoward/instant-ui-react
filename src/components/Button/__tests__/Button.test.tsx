import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Button } from "../Button";
import { ButtonProps } from "../Button.types";

const mockProps: ButtonProps = {
  text: "Example",
  disabled: false,
  primary: true,
  onClick: () => null,
};

describe("App", function () {
  it("should display pass in number", function () {
    render(<Button {...mockProps} />);

    expect(screen.getByTestId("button")).toHaveTextContent("Example");
  });
});
