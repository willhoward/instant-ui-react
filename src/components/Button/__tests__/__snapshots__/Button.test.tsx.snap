// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should display pass in number 1`] = `
<div>
  <button
    class="sc-bcPKhP kQiXYg"
    priority="primary"
    type="button"
  >
    Example
  </button>
</div>
`;
